seacrowd/sea_datasets/melayu_standard_lisan/melayu_standard_lisan.py

```diff
@@ -0,0 +1,157 @@
+    "kl201731": "https://raw.githubusercontent.com/matbahasa/Melayu_Standard_Lisan/master/KL201731.txt",
+    "kl201732": "https://raw.githubusercontent.com/matbahasa/Melayu_Standard_Lisan/master/KL201732.txt",
+    "kl201733": "https://raw.githubusercontent.com/matbahasa/Melayu_Standard_Lisan/master/KL201733.txt",
+}
+
+_SUPPORTED_TASKS = [Tasks.SELF_SUPERVISED_PRETRAINING]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class MelayuStandardLisan(datasets.GeneratorBasedBuilder):
+    """Korpus Variasi Bahasa Melayu:
+    Standard Lisan is a language corpus sourced from monologues of various melayu folklores."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = TASK_TO_SCHEMA[_SUPPORTED_TASKS[0]].lower()
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=f"{_DATASETNAME}",
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=f"{_DATASETNAME}",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_ssp":
+            features = schemas.self_supervised_pretraining.features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            # citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        urls = [_URLS[key] for key in _URLS.keys()]
+        data_path = dl_manager.download_and_extract(urls)
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": data_path[0], "split": "train", "other_path": data_path[1:]},
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str, other_path: List) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        filepaths = [filepath] + other_path
+        data = []
+        for filepath in filepaths:
+            with open(filepath, "r") as f:
+                data.append([line.rstrip() for line in f.readlines()][0])
```

Comment:
We take the first sentence for all the files. Is this an expected behavior? 

Right now the output only has one row, which is the concatenated text of the first sentences for each of the 33 files. 

```
>>> data = load_dataset("seacrowd/sea_datasets/melayu_standard_lisan/melayu_standard_lisan.py", name="melayu_standard_lisan_seacrowd_ssp")

>>> data
DatasetDict({
    train: Dataset({
        features: ['id', 'text'],
        num_rows: 1
    })
})
```

Reply:
Sorry for the mistake. The correct implementation should be: concatenating all the sentences in each file instead of only taking the first sentence. Have revised and committed. please have a check.